### ISSUE TITLE
Item bugging issue with recent update (0.5.8)

### ISSUE THREAD

**Issue by gothgirlwife** [MSchmoecker/No-Chest-Block#11]
Since the recent update, if my friends are using the same chest at the same time and we happen to interact with an item in the chest close enough together (and it doesn't always have to be the same item, but we could not figure out what caused that particular differentiation in our replication tests. Some of us think it might be item proximity in the box? But we are unsure), the item bugs out and several things happen:

The item I move to my inventory can no longer be selected or moved around.
1. I am unable to drop items from my inventory.
2. Any items I attempt to pick up from the ground manually vanish from the world and do not go in my inventory.
3. The item (sometimes, testing seems to indicate this happens when the same item is interacted with at the same time or in a similar time frame) stays in the chest and can no longer be moved.
4. Occasionally, we can no longer drop any items at all from our inventory once issue #1 occurs (Unsure what causes this. It happens sometimes immediately, sometimes it takes a few minutes to happen, and sometimes it doesn't happen at all).

Once we all quit the game and start it back up again, the issues are fixed with a couple of unfortunate side effects:

A) The item from issue #1 vanishes from my inventory and thus the world
B) Any items picked up during issue #2 are permanently gone from the world (once we figured out this was happening we could avoid it. However it does mean the first time we encountered this bug we lost about 150 scrap iron)

We did our best to figure out how to replicate these issues in the hope that it helps you solve them! Thank you so much for making such a useful mod, our game experience would def be a lot more frustrating without it.

**Comment by MSchmoecker** [comment]
Thanks for the report, can you please send your log files from BepInEx/LogOutput.log here after the issue occurred? I'll try to reproduce it but it may be a bug that only happens in combination with another mod.

**Comment by gothgirlwife** [comment]
[LogOutput.log](https://github.com/MSchmoecker/No-Chest-Block/files/14456906/LogOutput.log)

Here's the log! I didn't see any errors specifically tied to a mod specifically in the bepinex window before I closed, but maybe you'll see something I didn't!

**Comment by MSchmoecker** [comment]
This is the error:
```
[Info   : Unity Log] 02/29/2024 23:49:40: Exception in ZRpc::HandlePackage: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.NullReferenceException: Object reference not set to an instance of an object
  at MultiUserChest.ContainerRPCHandler.HandleRPC[TResponse,TInput] (ZDOID containerId, System.Int64 target, System.String rpcInvoke, System.Func`3[T1,T2,TResult] message, TInput input) [0x000d9] in D:\Dokumente\CodeProjects\ValheimProjects\NoChestBlock\MultiUserChest\ContainerRPCHandler.cs:55 
  at MultiUserChest.ContainerRPCHandler.RPC_RequestItemRemove (System.Int64 sender, ZDOID containerId, ZPackage package) [0x00000] in D:\Dokumente\CodeProjects\ValheimProjects\NoChestBlock\MultiUserChest\ContainerRPCHandler.cs:13 
  at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0006a] in <834b2ded5dad441e8c7a4287897d63c7>:0 
   --- End of inner exception stack trace ---
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00083] in <834b2ded5dad441e8c7a4287897d63c7>:0 
  at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) [0x00000] in <834b2ded5dad441e8c7a4287897d63c7>:0 
  at System.Delegate.DynamicInvokeImpl (System.Object[] args) [0x000e7] in <834b2ded5dad441e8c7a4287897d63c7>:0 
  at System.MulticastDelegate.DynamicInvokeImpl (System.Object[] args) [0x00008] in <834b2ded5dad441e8c7a4287897d63c7>:0 
  at System.Delegate.DynamicInvoke (System.Object[] args) [0x00000] in <834b2ded5dad441e8c7a4287897d63c7>:0 
  at RoutedMethod`2[T,U].Invoke (System.Int64 rpc, ZPackage pkg) [0x0001d] in <dbd2a6fbcde9498cadcacfb37ef883e3>:0 
  at ZRoutedRpc.HandleRoutedRPC (ZRoutedRpc+RoutedRPCData data) [0x00022] in <dbd2a6fbcde9498cadcacfb37ef883e3>:0 
  at ZRoutedRpc.RPC_RoutedRPC (ZRpc rpc, ZPackage pkg) [0x00023] in <dbd2a6fbcde9498cadcacfb37ef883e3>:0 
  at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0006a] in <834b2ded5dad441e8c7a4287897d63c7>:0 
   --- End of inner exception stack trace ---
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00083] in <834b2ded5dad441e8c7a4287897d63c7>:0 
  at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) [0x00000] in <834b2ded5dad441e8c7a4287897d63c7>:0 
  at System.Delegate.DynamicInvokeImpl (System.Object[] args) [0x000e7] in <834b2ded5dad441e8c7a4287897d63c7>:0 
  at System.MulticastDelegate.DynamicInvokeImpl (System.Object[] args) [0x00008] in <834b2ded5dad441e8c7a4287897d63c7>:0 
  at System.Delegate.DynamicInvoke (System.Object[] args) [0x00000] in <834b2ded5dad441e8c7a4287897d63c7>:0 
  at ZRpc+RpcMethod`1[T].Invoke (ZRpc rpc, ZPackage pkg) [0x0001d] in <dbd2a6fbcde9498cadcacfb37ef883e3>:0 
  at (wrapper dynamic-method) ZRpc.DMD<ZRpc::HandlePackage>(ZRpc,ZPackage)
  at ZRpc.Update (System.Single dt) [0x0003e] in <dbd2a6fbcde9498cadcacfb37ef883e3>:0 
```

Looks like this are ships and carts, have you interacted with them?

**Comment by gothgirlwife** [comment]
Ah, I didn't even realize it only happened with ships and carts! But looking back now, yeah it did only happen near ships and carts, not with boxes specifically.

**Comment by MSchmoecker** [comment]
Fixed in 0.5.9
